Sources/NIOOpenSSL/PosixPort.swift

```diff
@@ -27,7 +27,9 @@ private let sysMlock: @convention(c) (UnsafeRawPointer?, size_t) -> CInt = mlock
 private let sysMunlock: @convention(c) (UnsafeRawPointer?, size_t) -> CInt = munlock
 
 // Sadly, stat has different signatures with glibc and macOS libc.
-#if os(Linux) || os(FreeBSD) || os(Android)
+#if os(Android)
+private let sysStat: (UnsafePointer<CChar>, UnsafeMutablePointer<stat>) -> CInt = stat(_:_:)
```

Comment:
Hmm, this is the same signature as on Linux below, no?

Reply:
@weissi It looks like some strange Android issue, because with `@convention(c)` modificator it says:

```
PosixPort.swift:31:98: error: 
C function pointer signature 
'(UnsafePointer<Int8>!, UnsafeMutablePointer<stat>!) -> Int32' is not compatible with expected type '@convention(c) (UnsafePointer<CChar>, UnsafeMutablePointer<stat>) -> CInt' (aka '@convention(c) 
(UnsafePointer<Int8>, UnsafeMutablePointer<stat>) -> Int32')
```

So I removed `@convention(c)` modification.

Reply:
@albertaleksieiev ok, so the `@convention(c)` shouldn't have any effect on the type, it just means that it will be a C function rather than a Swift closure (and therefore faster). I think you might need to make this (adding the `!`s)

    @convention(c) (UnsafePointer<CChar>!, UnsafeMutablePointer<stat>!) -> CInt

Reply:
@weissi oh I see, thanks, but unfortunately `!` mark not allowed at the top level e.g. in arguments. But making it the same as in Darwin fixed it, e.g. `@convention(c) (UnsafePointer<CChar>?, UnsafeMutablePointer<stat>?) -> CInt`.